x-pack/plugins/security/server/routes/api/public/roles/put.js

```diff
@@ -84,30 +79,30 @@ export function initPutRolesApi(
     }, identity);
   };
 
-  const getGlobalSchema = () => {
+  const getKibanaSchema = () => {
     const privileges = authorization.privileges.get();
-    const featureObject = Object.entries(privileges.features).reduce((acc, [feature, featurePrivileges]) => ({
-      ...acc,
-      [feature]: Joi.array().items(Joi.string().valid(Object.keys(featurePrivileges)))
-    }), {});
-    const featureSchema = Joi.object(featureObject);
-    return Joi.object({
-      minimum: Joi.array().items(Joi.string().valid(Object.keys(privileges.global))),
-      feature: featureSchema,
+    const allSpacesSchema = Joi.array().length(1).items(Joi.string().valid([GLOBAL_RESOURCE]));
+    return Joi.array().items(
+      Joi.object({
+        base: Joi.alternatives().when('spaces', {
+          is: allSpacesSchema,
+          then: Joi.array().items(Joi.string().valid(Object.keys(privileges.global))),
+          otherwise: Joi.array().items(Joi.string().valid(Object.keys(privileges.space))),
+        }),
+        feature: Object.entries(privileges.features).reduce((acc, [feature, featurePrivileges]) => ({
```

Comment:
Does this need to be a `Joi.object()` rather than a native object? I remember you going through that exercise for other route validations recently

Reply:
Great catch, I will make it so.